getLocale.js

```diff
@@ -0,0 +1,13 @@
+module.exports = function getLocale (options) {
+  return new Promise((resolve, reject) => {
+    options.uv.on(/^([^.]+\.)?[a-z]{2}View$/, function (event) {
+      var { type, language, currency } = event
+      if (/^(basket|checkout|transaction)$/i.test(type)) return
```

Comment:
Why? Doesn't this mean that this promise will never resolve on basket/checkout/transaction pages, and thus we will never fetch recs?

Reply:
My understanding was the `replay()` will replay all recent events and we basically skip over those... code is from Aura so I assume they had a good reason for that (maybe that language and currency are not set for those?)

Reply:
Yes, that is what happens. I copied from aura-widget... may not be necessary.

The other question is what happens if a page doesn't have _any_ ecView events... then this module's request will just hang.

Reply:
@philippQubit it will replay all view events in the current page load. Most of our clients are not single page applications, so there is only ever 1 view event per page load.

@alexjesp If you don't have any view events, everything is fucked, and you have bigger problems.

Reply:
I just looked at the aura code. That function you linked is their view activation logic. We don't fire aura on those page types, hence why they ignore those page types. The purpose of your function is very different - you only want the locale, so you don't give a crap about page type.

Reply:
Cool, I'll make amendments now